src/main/java/org/opentripplanner/routing/graph/GraphIndex.java

```diff
@@ -65,7 +60,7 @@
     public final Map<String, TripPattern> patternForId = Maps.newHashMap();
     public final Map<Stop, TransitStop> stopVertexForStop = Maps.newHashMap();
     public final Map<Trip, TripPattern> patternForTrip = Maps.newHashMap();
-    public final Multimap<Agency, TripPattern> patternsForAgency = ArrayListMultimap.create();
+    public final Map<String, Multimap<Agency, TripPattern>> agencyPatternsForFeedId = Maps.newHashMap();
```

Comment:
It looks like the TripPatterns are still organized by agency. Because they are based on TripIds which are feed-unique I think this could just be a Multimap<String, TripPattern>.